Tensile/kernelConverter.py

```diff
@@ -0,0 +1,317 @@
+    currentIndexProblemType = currentIndexSolution["ProblemType"]
+    
+    #Iterate over problem type parameters and form the Yaml data
+    FormProblemTypeYamlData(currentIndexProblemType,versionString)
+    if args.skipMI != True:
+        #Iterate over Fork  parameters and form the Yaml data
+        FormForkParametersYamlData(currentIndexSolution, MIInstruction9Bits)
+    else:
+        print1("--skipMI option specified so skipping the Matrix Instruction parameter ...\n")   
+        FormForkParametersDataForNonMI(currentIndexSolution)
+    
+    #Forms the Library logic string
+    FormProblemSizeYamlData(exactLogic,solutionIndex,scheduleName,deviceNames,architectureName)
+    
+    #Write the Formed Yaml data into the Yaml File 
+    WriteToTensileYamlFile(TensileYamlFile)
+    
+def checkPositive(value):
+    try:
+        value = int(value)
+        if value <= 0:
+            raise RuntimeError("{} is not a positive integer".format(value))
+            
+    except ValueError:
+        raise Exception("{} is not an integer".format(value))
+    return value
+
+def FormProblemTypeYamlData(currentIndexProblemType, versionString):
+    if len(currentIndexProblemType) == 0:
+       raise RuntimeError("Length of problem Type Parameters is empty!!, Please re-check the library logic file !")
+    
+    # Form Global parameters section as well    
+    YamlFileData.append("GlobalParameters:\n")
+    YamlFileData.append("  MinimumRequiredVersion: {}\n\n".format(versionString['MinimumRequiredVersion']))
+    YamlFileData.append("BenchmarkProblems:\n  -\n    - # ProblemType\n")       
+
+    for problemTypeKey, problemTypeValue in currentIndexProblemType.items():
+     # always print DataType, DestDataType, ComputeDataType, HighPrecisionAccumulate, OperationType fields
+      if problemTypeKey == "DataType" or problemTypeKey == "DestDataType" or problemTypeKey == "ComputeDataType":
+         YamlFileData.append("      {}: {}\n".format(problemTypeKey, dataType[problemTypeValue]))
+         continue
+      if problemTypeKey == "HighPrecisionAccumulate" or problemTypeKey == "OperationType":
+         YamlFileData.append("      {}: {}\n".format(problemTypeKey, problemTypeValue))
+         continue
+      if problemTypeKey in defaultProblemType:
+         if problemTypeValue != defaultProblemType[problemTypeKey]:
+            YamlFileData.append("      {}: {}\n".format(problemTypeKey, problemTypeValue))
+            
+def FormForkParametersDataForNonMI(currentIndexSolution):
+
+    YamlFileData.append("    - # BenchmarkProblemSizeGroup\n") 
+    YamlFileData.append("      InitialSolutionParameters:\n") 
+    YamlFileData.append("      BenchmarkCommonParameters:\n") 
+    
+    YamlFileData.append("      ForkParameters:\n")
+    macInstruction = ""
+    for forkKey, forkValue in currentIndexSolution.items():
+      if forkKey == "MACInstruction":
+        if forkValue == "MAC": #This is a bug in the library logic file

```

Comment:
Are you sure about "This is a bug"?

Reply:
Hi @benjaminulmer 
Do you know why we have MACInstruction: MAC in rocblas logic file?

Reply:
@nakajee My answer to Babak previously on this matter:
> I don't have a specific answer, but there are a lot of ways parameter values can make it into rocBLAS that aren't "valid" values as specified in Common.py. Usually what happens is they get set that way in SolutionStructs for some reason or another

> and that might have happened in an older version of Tensile where that no longer happens anymore